### PR TITLE
patch(palette): storybook with rendering outside the safe area

### PR DESCRIPTION
**Pull request by pvinis**
This PR resolves []

### Description

This is an addon PR for #6495. It upgrades to storybook 6, to make patching easier, and adds a patch, which is a recreation of https://github.com/storybookjs/react-native/pull/345, to allow for Screen stories rendering. Look at that PR for screenshots.

### PR Checklist (tick all before merging)

- [ ] Tested on **iOS** and **Android**
- [x] Screenshots and videos 
- [x] Added Tests and Stories
- [ ] App state migration
- [ ] Feature flag

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

### CODE REVIEW

File: src/palette/atoms/Spacer/Spacer.stories.tsx [hunk 1]
```diff
@@ -48,7 +47,6 @@ function SpacerCol(props: { y: SpacerProps["y"] }) {
 }
 
 storiesOf("palette/atoms/Spacer", module)
-  .addDecorator(withTheme)
```
Comment (pvinis) [inline src/palette/atoms/Spacer/Spacer.stories.tsx:51]: storybook 6 fixed that finally! no need for these anymore

File: metro.config.js [hunk 2]
```diff
@@ -15,4 +15,8 @@ module.exports = {
       },
     }),
   },
+
+  resolver: {
+    resolverMainFields: ["sbmodern", "react-native", "browser", "main"], // needed for storybook
```
Comment (araujobarret) [inline metro.config.js:20]: Are we sure we're not bundling Storybook in release builds or we were doing that before and we're fine with it?

**Review by araujobarret — APPROVED** [review]
A super one 🎖️

File: metro.config.js (hunk 2 above)
Comment (pvinis) [reply metro.config.js:20]: we are fine. we want to try and enable it from the dev menu.

File: EXAMPLES.md [hunk 3]
```diff
@@ -82,11 +82,11 @@ Links:
 
 We use `Screen` and friends for that. They come from palette. Some of the building blocks are: `Screen`, `Screen.Body`, `Screen.Header`, `Screen.FloatingHeader`, `Screen.Background`, `Screen.BottomView`, `Screen.BodyPadding`.
 
-You can find examples in [Screen.fakestories.tsx](src/palette/organisms/screenStructure/Screen.fakestories.tsx). You can also find actual screens using these, like  [OnboardingWelcome.tsx](src/app/Scenes/Onboarding/OnboardingWelcome.tsx),  [OnboardingSocialPick.tsx](src/app/Scenes/Onboarding/OnboardingSocialPick.tsx),  [OnboardingLogin.tsx](src/app/Scenes/Onboarding/OnboardingLogin.tsx).
+You can find examples in [Screen.stories.tsx](src/palette/organisms/screenStructure/Screen.stories.tsx). You can also find actual screens using these, like [OnboardingWelcome.tsx](src/app/Scenes/Onboarding/OnboardingWelcome.tsx), [OnboardingSocialPick.tsx](src/app/Scenes/Onboarding/OnboardingSocialPick.tsx), [OnboardingLogin.tsx](src/app/Scenes/Onboarding/OnboardingLogin.tsx).
 
 Links:
 
-- [Screen.fakestories.tsx](src/palette/organisms/screenStructure/Screen.fakestories.tsx)
+- [Screen.stories.tsx](src/palette/organisms/screenStructure/Screen.stories.tsx)
```
Comment (brainbicycle) [inline EXAMPLES.md:89]: ✨

File: src/palette/organisms/screenStructure/Screen.stories.tsx [hunk 4]
```diff
@@ -1,18 +1,159 @@
...
-          To test our `Screen`, use `yarn start` and navigate to `Screen.fakestories.tsx` in your
-          editor.
-        </Text>
-      </Screen.Body>
-    </Screen>
-  ))
+const ScreenMeta: ComponentMeta<typeof Screen> = {
+  title: "palette/organisms/Screen",
+  component: Screen,
+  parameters: { noSafeArea: true },
```
Comment (brainbicycle) [inline src/palette/organisms/screenStructure/Screen.stories.tsx:8]: nice

File: package.json [hunk 5]
```diff
@@ -107,9 +107,11 @@
     "@react-native-async-storage/async-storage": "1.16.1",
     "@react-native-community/cameraroll": "1.3.0",
     "@react-native-community/clipboard": "1.5.1",
+    "@react-native-community/datetimepicker": "6.1.2",
```
Comment (brainbicycle) [inline package.json:110]: why do we need these? 🤔

**Review by brainbicycle — APPROVED** [review]

File: package.json (hunk 5 above)
Comment (pvinis) [reply package.json:110]: these were needed by storybook. I forgot to add a comment about this here.

File: EXAMPLES.md (hunk 3 above)
Comment (pvinis) [reply EXAMPLES.md:89]: thanks for the bump to do it right 💜. I def agree with it, but upgrade fatigue got me 😅.

**Pull request merged**